pennylane/debugging.py

```diff
@@ -76,16 +116,79 @@ def snapshots(qnode):
         def circuit():
             qml.Snapshot(measurement=qml.expval(qml.Z(0))
             qml.Hadamard(wires=0)
-            qml.Snapshot("very_important_state")
+            qml.Snapshot("counts_checks", qml.counts(), shots=100)
             qml.CNOT(wires=[0, 1])
             qml.Snapshot()
             return qml.expval(qml.X(0))
 
     >>> qml.snapshots(circuit)()
     {0: 1.0,
-    'very_important_state': array([0.70710678+0.j, 0.        +0.j, 0.70710678+0.j, 0.        +0.j]),
+    'counts_check': {"00": 55, "10": 45},
     2: array([0.70710678+0.j, 0.        +0.j, 0.        +0.j, 0.70710678+0.j]),
     'execution_results': 0.0}
+
+    .. code-block:: python3
+
+        @qml.snapshots
+        @qml.qnode(qml.device("lightning.qubit", wires=2), diff_method="parameter-shift")
+        def circuit():
+            qml.Hadamard(wires=0),
+            qml.Snapshot("very_important_state"),
+            qml.CNOT(wires=[0, 1]),
+            return qml.expval(qml.PauliZ(0))
+
+        with circuit.device.tracker:
+            circuit()
+
+    >>> circuit.device.tracker.totals
+    {'batches': 1, 'simulations': 2, 'executions': 2}
+    """
+    new_tapes = []
+    accumulated_ops = []
+    snapshot_tags = []
+
+    for op in tape.operations:
+        if isinstance(op, qml.Snapshot):
+            snapshot_tags.append(op.tag or len(new_tapes))
+            shots = op.hyperparameters["shots"]
+
+            if shots == -1:
+                shots = tape.shots
+
+            shots = (
+                qml.measurements.Shots(shots)
+                if not isinstance(shots, qml.measurements.Shots)
+                else shots
+            )
```

Comment:
```suggestion
            shots = qml.measurements.Shots(shots)
```

Calling `Shots` on a shots object just returns the original shots object.